### DISCUSSION TITLE
Use Vitess as backed data storage: MySQL compatibility issues

### DISCUSSION THREAD

**Discussion by xuwang-wish** [ory/hydra#2712] [Q&A] [answered]
Hi,

We are currently using AWS RDS (MySQL mode) as data source name in Hydra config, but we are planning to migrate to [Vitess](https://vitess.io/).  

We noticed that Vitess has some [compatibility issues with MySQL](https://vitess.io/docs/reference/compatibility/mysql-compatibility/), so,

is Hydra now using any of those mentioned features? 
If yes, in what scenarios? what will be the impact if we migrate to Vitess?

Thank you.

**Comment by Benehiko** [comment, marked as answer]
Hi @xuwang-wish,

I see this is similar to this slack thread [here](https://ory-community.slack.com/archives/C012RBW0F18/p1630618493008500).

I haven't worked with Vitess before, nor have I tried running Hydra with it so I can't speak out of experience. 
It is also quite difficult to evaluate which features are missing since there are many queries that would need to be evaluated. We currently run tests against PostgreSQL, MySQL, CockroachDB and SQLite to ensure any query written will work for them. We also use a third-party library called [go-buffalo pop](https://gobuffalo.io/en/docs/db/getting-started/) which interfaces for us to each of the listed databases. That is why I cannot guarantee Vitess to work currently or in future. The best would be to try it out. 

Another option is to open an issue to support Vitess.

I hope this explanation helps.

**Reply by vinckr** [reply]
Hey @xuwang-wish ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
